### PR TITLE
Update Player.java, and simplified functions

### PR DESCRIPTION
**Pull request by stalight**
Jillian should especially check because I left some notes for some functions which you may need to take reference of when implementing items

### CODE REVIEW

**Review by littlepeach17 — DISMISSED** [review]
Looks good and yep I've taken note about the Item stuff

**Review by littlepeach17 — APPROVED** [review]
Implements Triggerable is not needed for subclasses of Event, because Event already implements Triggerable

**Review by jchenDDD — APPROVED** [review]

**Review by UTDQi — APPROVED** [review]

**Comment by villanueva-j** [comment]
I have some comments and questions we can discuss on the discord.

for Player class:
- Since we already discussed creating an inventory class, for "weapon" and "armor" in inventory, I suggest we create specific weapon and armor classes, and set the type of those parameters to each respective class (ex: Weapon weapon, Armor armor). That way we have a way to prevent armor equipment from being assigned to weapon and vice versa.
- type difference for method getPlayerLocation vs. location type (method return hashmap<Character, Integer> but location is of type <String, Integer>)

for Player constructor: 
- the code can be made cleaner by creating an inventory class like you said 
- Does Player always start with same base stats? If so, we can overload the constructor to have default stats.
- Does Player start with any equipment? then for inventory class, we can have a constructor that does not assign anything to weapon and armor and one that does to allow for extension for if Players are given a choice of weapon

for Events:
- looks good. Maybe if we have time we can add some sort of cute text that is returned when a player triggers WallEvent like "Ouch, that's a wall". 
 
You should add javadoc for all your methods so it is clear what each does. Other than that looks good!

**Review by littlepeach17 — APPROVED** [review]

File: src/main/java/entity/CollectibleInventory.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        return this.inventory.get(collectibleType);
+    }
+
+    public void setInventory(String collectibleType, int amount){
+        /*
+        Precondition: - Gets the collectibleType (in this case Essence or Artifact)
+                      - The amount is -inf < amount < inf
+        Postcondition: - Changes the amount in collectible
+         */
+        Collectible collectible = getCollectible(collectibyType);
```
Comment (villanueva-j) [inline src/main/java/entity/CollectibleInventory.java:30]: typo here

**Review by villanueva-j — CHANGES_REQUESTED** [review]
Typo fix and add in javadoc

Some methods you put javadoc in but you put it after method declaration, it needs to go before

**Review by jchenDDD — DISMISSED** [review]

**Review by littlepeach17 — DISMISSED** [review]
documentation looks nice

**Comment by stalight** [comment]
I changed the Interface
"Triggerable" to "Enterable"
and moved
"Trigger" to Event parent class

**Pull request merged**